src/app-layer-parser.c

```diff
@@ -1011,6 +1014,20 @@ int AppLayerParserParse(ThreadVars *tv, AppLayerParserThreadCtx *alp_tctx, Flow
         }
     }
 
+    if (AppLayerParserProtocolIsTxAware(f->proto, alproto)) {
+        if (likely(tv)) {
+            /* At each pass in the parser we add to the tv counter
```

Comment:
I don't see why you need it to be in the state.

Why would this not work?
<pre>
p_tx_cnt = ...GetTxCnt();
...
Parser Runs
...
cur_tx_cnt = ...GetTxCnt()
if (cur_tx_cnt > p_tx_cnt) {
    new_tx_cnt = cur_tx_cnt - p_tx_cnt;
}
</pre>